Libraries/Tools/Az/Create-InitiativeDef.ps1

```diff
@@ -1,81 +1,79 @@
 # region parameters
 param (
-    [Parameter(Mandatory=$true)][string]$rootFolder,
-    [Parameter(Mandatory=$true)][string[]]$modifiedInitiatives,
-    [Parameter(Mandatory=$true)][string]$managementGroupName
+    [Parameter(Mandatory = $true)][string]$rootFolder,
+    [Parameter(Mandatory = $true)][string[]]$modifiedInitiatives,
+    [Parameter(Mandatory = $true)][string]$managementGroupName
 )
 #endregion
 
-$azPD = Get-AzPolicyDefinition
+if ($modifiedInitiatives -eq $null) {
+    Write-Host "##vso[task.LogIssue type=warning;]No Initiative diffs found or only deleted files found, skiping..."
+}
+else {
+    # Inititaives list is not null or empty
+    $azPD = Get-AzPolicyDefinition -ManagementGroupName $managementGroupName
 
-foreach ($modifiedInitiative in $modifiedInitiatives) {
+    foreach ($modifiedInitiative in $modifiedInitiatives) {
 
-    $filePath = $rootFolder + "\Initiatives\" + $modifiedInitiative + ".json"
+        $filePath = $rootFolder + $modifiedInitiative
 
-    Write-Host "##[debug] File path: $filePath"
+        Write-Host "##[debug] File path: $modifiedInitiative"
 
-    $azureInitiative = Get-Content $filePath | ConvertFrom-Json
+        $azureInitiative = Get-Content $filePath | ConvertFrom-Json
 
-    Write-Host "##[debug] Policy $($azurePolicy.properties.displayName)"
+        Write-Host "##[debug] Initiative $($azureInitiative.properties.displayName)"
 
-    $policyDefinitions = @()
+        $policyDefinitions = @()
 
-    foreach ($policy in $azureInitiative.properties.PolicyDefinitions) {
+        foreach ($policy in $azureInitiative.properties.PolicyDefinitions) {
 
-        Write-Host "##[section] Processing $($policy.policyDefinitionName)"
+            Write-Host "##[section] Processing $($policy.policyDefinitionName)"
 
-        $policyLookup = $azPD | Where-Object {$_.Name -eq $policy.policyDefinitionName}
+            $policyLookup = $azPD | Where-Object { $_.Name -eq $policy.policyDefinitionName }
 
-        if($policyLookup){
-            Write-Host "##[debug] Policy found setting policyID"
+            if ($policyLookup) {
+                #Write-Host "##[debug] Policy found setting policyID"
```

Comment:
@nicko3185, are lines 35 and 36 meant to be commented out?

Reply:
We commented them out because they were generating a lot of unnecessary output, but they are helpful for debugging so we left them there.